liveRecorder/Tasks/ConcatinationTask.py

```diff
@@ -93,16 +95,43 @@ def parse_m3u8(m3u8):
 
     def run(self):
 
-        output_full_path = os.path.join(self.recording_path, self.output_filename)
-        if os.path.isfile(output_full_path):
-            self.logger.warn("file [%s] already exist", output_full_path)
-            return
+        command = self.ts_2_mp4_converter_path + ' '
         token = self.tokenize_url(self.token_url)
         self.url_base_entry = self.nginx_url.format(token)
         self.url_master = os.path.join(self.url_base_entry, 'master.m3u8')
-        url_source_manifest = self.find_source()
-        playlist = self.download_file(url_source_manifest)
-        self.logger.debug("load recording manifest : \n %s ", playlist)
-        chunks = m3u8.loads(playlist).files
-        self.download_chunks_and_concat(chunks, output_full_path)
-        self.logger.info("Successfully concat %d files into %s", len(chunks), output_full_path)
\ No newline at end of file
+        flavors_manifest_list = self.find_source()
+        prog = re.compile(self.flavor_pattern)
+
+
+        for url_source_manifest in flavors_manifest_list.values():
+            result = prog.match(url_source_manifest)
+            # check for errors
+            if not result:
+                # log error
+                error = "Error running concat task, failed to parse flavor from url: [%s]", url_source_manifest
+                self.logger.error(error)
+                raise ValueError(error)
+            ts_output_filename = self.get_output_filename(result.group('flavor'))
+            output_full_path = os.path.join(self.recording_path, ts_output_filename[0])
+            mp4_output_filename = ts_output_filename.split('.') + '.mp4'
+            if os.path.isfile(output_full_path):
+                self.logger.warn("file [%s] already exist", output_full_path)
+                return
+            playlist = self.download_file(url_source_manifest)
```

Comment:
you removed the tokenize_url code... without valid token it will fail in production